### PR TITLE
CI - Compilation options based on pull request labels

### PR DESCRIPTION
**Pull request by MegMll**
This pr aims at providing cmake compilations options based on the labels from pull requests. When no label are added, a minimal setup is provided. 

For more informations, please see https://github.com/stack-of-tasks/pinocchio/issues/2251

### CODE REVIEW

**Review by jcarpent — COMMENTED** [review]
Very nice action and idea. Thanks to @MegMll and @jorisv

File: .github/workflows/scripts/get_label.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            ' -DGENERATE_PYTHON_STUBS=ON'
+        ],
+        build_collision: ' -DBUILD_WITH_COLLISION_SUPPORT=ON',
+        build_casadi: ' -DBUILD_WITH_CASADI_SUPPORT=ON',
+        build_autodiff: ' -DBUILD_WITH_AUTODIFF_SUPPORT=ON',
+        build_codegen: ' -DBUILD_WITH_CODEGEN_SUPPORT=ON',
+        build_extra: ' -DBUILD_WITH_EXTRA_SUPPORT=ON',
+        build_openmp: ' -DBUILD_WITH_OPENMP_SUPPORT=ON',
+        build_mpfr: ' -DBUILD_PYTHON_BINDINGS_WITH_BOOST_MPFR_SUPPORT=ON',
+        build_sdf: ' -DBUILD_WITH_SDF_SUPPORT=ON'
```
Comment (jorisv) [inline .github/workflows/scripts/get_label.js:44]: - sdf: To manage sdf we have to install gazebo dependency. But this dependency can create a lot of conflict.
- Do we add build_urdf ? In this case, we have to deactivate it by default.
- We can add build_accerelate option. CMakeLists.txt set it up to FALSE on other platform than Mac so we don't have to add a test in this script.

**Review by jorisv — CHANGES_REQUESTED** [review]
Great work !

get_label.js allow to factorize a lot of code.

File: .github/workflows/scripts/get_label.js (hunk 1 above)
Comment (jcarpent) [reply .github/workflows/scripts/get_label.js:44]: We need gazebo for SDF?

Comment (MegMll) [reply .github/workflows/scripts/get_label.js:44]: For SDF, i tried and in the ned we only need libsdformat, which can be installed via conda. It's quite light, so might be worth it. 
And i added the accelerate option

Comment (jorisv) [reply .github/workflows/scripts/get_label.js]: My bad, I had some issue month ago to find sdf library on conda.

**Comment by jorisv** [comment]
This PR don't require a changelog entry since it only modify the CI.

**Review by jorisv — APPROVED** [review]

File: .github/workflows/scripts/get_label.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        else
+        {       
+            const reviewMessage = `👋 Hi,
+            this is a reminder message to please assign a proper label to this Pull Request.
+            The possible labels are:
+            
+            - build_collision (build pinocchio with coal support)
+            - build_casadi (build pinoochio with casadi support)
+            - build_autodiff (build pinocchio with cppad support)
+            - build_codegen
```
Comment (jorisv) [inline .github/workflows/scripts/get_label.js:88]:
```suggestion
            - build_codegen (build pinocchio with cppadcg support)
```

File: .github/workflows/scripts/get_label.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            - build_extra (build pinocchio with extra algorithms)
+            - build_mpfr
```
Comment (jorisv) [inline .github/workflows/scripts/get_label.js:90]:
```suggestion
            - build_mpfr (build pinocchio with Boost.Multiprecision support)
```

File: .github/workflows/scripts/get_label.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+            - build_sdf (build sdf parser)
```
Comment (jorisv) [inline .github/workflows/scripts/get_label.js:91]:
```suggestion
            - build_sdf (build pinocchio with sdf parser)
```

**Review by jorisv — CHANGES_REQUESTED** [review]

**Review by jorisv — APPROVED** [review]

**Pull request merged**